kong/router/atc_compat.lua

```diff
@@ -745,6 +748,11 @@ function _M:exec(ctx)
     end
 
     self.cache:set(cache_key, match_t)
+
+  else
+    if ctx then
```

Comment:
we can move this `if ctx then ... end` out of `if not match`, after `-- found a match`

Reply:
No. This is not the logic you think it is. Out of the branch, it can be both cached or not cached. Only inside the branch, it is cached.